modules/azure-public-dns/init.tf

```diff
@@ -0,0 +1,10 @@
+terraform {
```

Comment:
We don't require these inside modules. These will be set with the component that will be targeted to call a module and not module itself.

Reply:
what's wrong with having it here? this is just requiring whoever uses the module they must have at least these versions

Reply:
There is nothing wrong with having these inside the module. But the structure that we are targetting is to allow consistency over how we do terraform across the program. The structure to follow is that we won't be running the terraform from inside the module but we will be running it from the components. So the component will govern the use of terraform and provider versions and not the module. The module will be kept as generic as possible to allow reuse.

Reply:
also, we are governing the use of terraform versions by .terraform-versions file and the tfenv tool with which the person running the terraform manually will automatically use the desired version of the terraform and he wouldn't need to declare the version inside a module/